src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -205,6 +211,38 @@ private void ChangeProjectProperty<T>(ref T field, T newValue, Func<Solution, So
             }
         }
 
+        /// <summary>
+        /// Reports a telemetry event if compilation information is being thrown away after being previously computed
+        /// </summary>
+        private static void TryReportCompilationThrownAway(SolutionState solutionState, ProjectId projectId)
+        {
+            // We log the number of syntax trees that have been parsed even if there was no compilation created yet
+            var projectState = solutionState.GetRequiredProjectState(projectId);
+            var parsedTrees = 0;
+            foreach (var documentState in projectState.DocumentStates.Values)
```

Comment:
📝 Iterating `DocumentStates` is not free. This method should return early if telemetry is disabled.

Reply:
Sure thing. To ask the obvious question: How?

VSTelemetryLogger has an IsEnabled method, but its per function, always returns true, and I can't get access to the instance of it to check it anyway.

Reply:
You should be able to use `IWorkspaceTelemetryService.HasActiveSession` to check.